### ISSUE TITLE
Type errors are not reported, only syntax errors.

### ISSUE THREAD

**Issue by justinfagnani** [ivogabe/gulp-typescript#368]
**Expected behavior:**

gulp-typescript reports type errors - it should report the same errors that `tsc` does.

**Actual behavior:**

gulp-typescript doesn't report type errors. It does report syntax errors.

Also, using the `noEmitOnError: true` option does cause the task to fail, though the specific errors are not reported.

**Your gulpfile:**

```javascript
const tsProject = typescript.createProject('tsconfig.json');

gulp.task('build', () =>
  gulp.src('src/**/*.ts')
    .pipe(typescript(tsProject))
    .pipe(gulp.dest('lib'))
);
```

**tsconfig.json**

```json
{
  "compilerOptions": {
    "target": "es6",
    "module": "commonjs",
    "moduleResolution": "node",
    "isolatedModules": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "declaration": false,
    "noImplicitAny": true,
    "removeComments": false,
    "noLib": true,
    "preserveConstEnums": true,
    "suppressImplicitAnyIndexErrors": false,
    "outDir": "lib"
  },
  "filesGlob": [
    "custom_typings/*.d.ts",
    "src/**/*.ts",
    "!node_modules/**",
    "typings/index.d.ts",
    "node_modules/typescript/lib/lib.core.es6.d.ts"
  ],
  "files": [
    "custom_typings/dom5.d.ts",
    "custom_typings/fs-extra.d.ts",
    "custom_typings/hydrolysis.d.ts",
    "custom_typings/node.d.ts",
    "custom_typings/plylog.d.ts",
    "custom_typings/slash.d.ts",
    "custom_typings/vinyl-fs.d.ts",
    "src/analyzer.ts",
    "src/build.ts",
    "src/bundle.ts",
    "src/fork-stream.ts",
    "src/optimize.ts",
    "src/polymer-build.ts",
    "src/polymer-project.ts",
    "src/prefetch.ts",
    "src/project-config.ts",
    "src/streams.ts",
    "src/sw-precache.ts",
    "src/uglify-transform.ts",
    "src/url-from-path.ts",
    "typings/index.d.ts",
    "node_modules/typescript/lib/lib.core.es6.d.ts"
  ],
  "atom": {
    "rewriteTsconfig": true
  }
}
```

**Code**

I noticed this when I enabled `noImplicitAny` in my tsconfig, but it turns out there are other type errors reported by tsc and not by gulp-typescript. To check that gulp-typescript was actually catching any errors I introduced some syntax errors and those were indeed piped as errors and caused the task to fail.

I tried various combinations of using `tsProject.src()` instead of `fulp.src()`, using a custom Reporter (no errors are passed to it for type errors), and `noEmitOnError`. Only `noEmitOnError` caused anything to fail on type errors, or at least I presume. I need to fix my type errors to make sure emit works on error-free code.

**Comment by ivogabe** [comment]
With `isolatedModules`, all files are compiled separately. This improves compilation times a lot, but it does mean that type errors will not be reported (as all files are needed for the analysis). So you should either accept that or remove that option.

**Comment by justinfagnani** [comment]
Why does `tsc` report type errors with `isolatedModules`? How come implicit any errors can't be reported, since they're always local?

**Comment by justinfagnani** [comment]
Also, when I turn off `isolatedModules` (and move the `lib.core.es6.d.ts` to the top of `filesGlob` and `files`) `tsc` still works, but `gulp build` reports missing global types like `Array` that are usually an indication of not having a core lib declaration. I'm not using `noExternalResolve`, so this is unexpected.

**Comment by ivogabe** [comment]
gulp-typescript has as implementation that is very fast with `isolatedModules`. It compiles files one by one. When only one file changes, only that file is recompiled. But, with such implementation you cannot do type checking. I'm not sure how it was implemented in `tsc` but that's probably not as fast as gulp-typescript for large projects.

For your second issue, you're using `gulp.src` and you do not include the lib file there. You should list it there or use `tsProject.src()`, which will resolve files based on your tsconfig file.
